### DISCUSSION TITLE
Bairros de uma cidade por CEP

### DISCUSSION THREAD

**Discussion by jlucas577** [BrasilAPI/BrasilAPI#227] [Ideas]
Durante o desenvolvimento de um projeto pessoal, dei de cara com uma grande dificuldade em trabalhar com o sistema de CEPs, algumas cidades possuem apenas um CEP para tudo (Exemplo: a cidade de Paraty - 23970000) e alguns bairros pequenos não possuem CEP. Acredito que seria interessante retornar a lista de bairros de uma determinada cidade através da busca pelo CEP.

**Comment by HilSantos** [comment]
É uma boa ideia, também já passei por isso, quando vendia planos de Internet pelas redes sociais.

**Comment by wferreiracosta** [comment]
Gostei da ideia, na minha região houve uma troca de CEP e isso dificulta bastante uma implementação dessa seria muito útil.

**Comment by schirrel** [comment]
Essa ideia me interessa bastante também, me de parei com o problema de listagem de bairros na cidade que moro. Não existe nenhuma listagem atualizada, o máximo que tenho são listas antigas ou um 'olhometro' pelos bairros que constam no maps. 
Como vocês imaginam a aquisição das informações dos bairros?

**Comment by RodriAndreotti** [comment]
Então pessoal, mas será que existe este tipo de recurso?
Pergunto pois só vi dois casos distintos de CEP até hoje:
CEP para cidade, onde o CEP é único para a cidade inteira, alguns exemplos: 09450-000 - Rio Grande da Serra ou 18520-000 Cerquilho.

No site dos correios, quando buscamos a faixa de CEPs de uma localidade, para estes dois exemplos aparece a seguinte mensagem:
![cep](https://user-images.githubusercontent.com/9449048/87790307-c0573d80-c816-11ea-97b0-ff573cd3ada5.PNG)

Porém, o exemplo dado acima, da cidade de Paraty (que é o mesmo caso dos que citei) não é que tem bairros que não tem CEP, e sim que a cidade inteira usa o mesmo CEP, nas minha implementações, o que faço é manter um campo de controle na tabela de endereços dos cadastros que informa se o CEP é para a cidade ou para logradouro, e configuro este campo caso o retorno do api que estiver usando contenha cidade e logradouro, se tiver os dois, marco o campo como false, se tiver só cidade marco o campo de controle como true.

Mas creio que retornar a lista de todos os bairros, creio que o melhor seria utilizar informações do IBGE, não lembro na Brasil API, mas em alguns outros API eles mandam junto com a response o código IBGE da cidade, com isso seria possível retornar de algum outro api que consuma dos dados do IBGE a lista de bairros.

**Comment by schirrel** [comment]
@RodriAndreotti pode ser que ocorreu entendimento diverso sobre a necessidade, no meu caso quis dizer de uma API, que ao informar o CEP da Cidade, o cep geral, se retorna a lista dos bairros da cidade.

A questão de comentou sobre o IBGE, poderia se ter um cruzamento de informações onde pela api dos correios se pega a cidade por cep e se obtendo essa cidade se buscaria os bairros na api do ibge, desconheço se o IBGE trabalha com CEP também

**Comment by RodriAndreotti** [comment]
Fala, @schirrel !
Provavelmente foi isso mesmo, rs.
Sobre o IBGE e o CEP, como tudo no Brasil, não utiliza o CEP como referência, tem um código próprio, de cidade, para os exemplos acima seriam:
3544103 - Rio Grande da Serra
3511508 - Cerquilho
3303807 - Paraty

Mas sinceramente não sei se o IBGE teria também esta listagem de bairros, é o lugar mais propício para encontrar, e tem uma API bem rica (estável, e grátis... shame on you CORREIOS)
Teria que dar uma vasculhada na documentação deles para ver o que dá para encontrar por lá.
Esqueci de deixar o link, mas tem muita informação interessante em API's do IBGE:
[https://servicodados.ibge.gov.br/api/docs](https://servicodados.ibge.gov.br/api/docs)

**Comment by mechamogeo** [comment]
Fala Pessoal!

Eu trabalho na área de frete da @b2wdigital e lá utilizamos a base dos Correios, Waze, Google Maps, Nomenatim e etc ... com isso dividimos toda a base em região, micro região, meso região e mais algumas coisas ... Porém trouxe uma idéia olhando essa informações de bairros, não sei se entra no tópico certo, acredito que sim, mas juguem ...

Em determinado dia, resolvemos recolher a latitude e longitude, para algumas soluções, então olhando essa discussão pensei em algo diferente ... não é legal tentar pegar um cep próximo da pessoa, usando o gps como exemplo? realmente para saber se alí próximo não teria um outro cep cadastrado? Bom pra validar tentei fazer pegando a rua em que estou, porém como não estou em uma cidade sem quebra de cep pode dar certo, mas vou deixar aqui meu exemplo, se quiserem explorem:

> Busca pela latitude e longitude algum cep cadastrado pela proximidade (máximo 5 ceps)
> https://cep-v2-americanas-npf.b2w.io/cep?latitude=-22.8653529&longitude=-47.1931597

> Busca o cep que foi retornado na lista, que é um endereço que eu reconheço
> https://cep-v2-americanas-npf.b2w.io/cep/13184752

Bom assim você valida pela proximidade da pessoa e não pelo cep em sí, como disse anteriormente não sei se é o tópico certo mas acho que pode dar mais idéias ...

* Bom esse link, vocês podem pegar usando o inspecionar do Chrome então kkk não acho que tenha problema utilizar em algum projeto assim que seja poucas requisições kkk

**Comment by elirweb** [comment]
Boa tarde,
Em Mongaguá - SP  tem muitos casos disso. é uma ideia interessante e para complementar a ideia trazer a latitude e longitude, como se a pessoa além de buscar o "Cep" da localidade e poder encontrar outros lugares ao redor.

**Comment by HilSantos** [comment]
Ótimo! Ainda estou engatinhando no Github, se eu tiver uma ideia, ajudarei.
________________________________
De: elirweb <notifications@github.com>
Enviado: segunda-feira, 16 de novembro de 2020 17:16
Para: BrasilAPI/BrasilAPI <BrasilAPI@noreply.github.com>
Cc: Hilton Santos <hilvatos@hotmail.com>; Comment <comment@noreply.github.com>
Assunto: Re: [BrasilAPI/BrasilAPI] Bairros de uma cidade por CEP (#86)


Boa tarde,
Em Mongaguá - SP tem muitos casos disso. é uma ideia interessante e para complementar a ideia trazer a latitude e longitude, como se a pessoa além de buscar o "Cep" da localidade e poder encontrar outros lugares ao redor.
